MaxText/train.py

```diff
@@ -166,8 +167,28 @@ def clear_buffered_metrics():
   _buffered_step = None
   _buffered_metrics = None
 
-def save_checkpoint(checkpoint_manager, step, state, dataset_type="c4", data_iterator=None):
-  """Wrapper for saving checkpoint"""
+def save_checkpoint(
```

Comment:
Hmm not specific to this PR but this method should really be defined inside of checkpointing instead of train...